### PR TITLE
complete the cluster up struct in the Complete method

### PR DESCRIPTION
**Pull request by deads2k**
There are many "tasks" added the Complete method which actually mutate struct state and effectively complete the object.  This updates the method to simply call the checks and struct completion directly instead of adding them to a list to later be run.

This allows the values to be safely used after calling complete like other CLI commands.

/assign @mfojtik 

part of cluster up pull.

### CODE REVIEW

File: pkg/oc/bootstrap/docker/up.go [hunk 1]
```diff
@@ -350,51 +350,89 @@ func (c *CommonStartConfig) Complete(f *osclientcmd.Factory, cmd *cobra.Command,
...
-		taskPrinter.Success()
+	taskPrinter.StartTask("Create Docker client")
+	client, err := getDockerClient(out, c.DockerMachine, true)
+	if err != nil {
+		return taskPrinter.ToError(err)
 	}
+	c.dockerClient = client
+	taskPrinter.Success()
+
+	// TODO HOLY SIDE-EFFECTS this is setting global bool values as part of a check!
```
Comment (mfojtik) [inline pkg/oc/bootstrap/docker/up.go]: lol

File: pkg/oc/bootstrap/docker/up.go [hunk 2]
```diff
@@ -1180,21 +1213,6 @@ func (c *ClientStartConfig) InstallMetrics(out io.Writer) error {
 	return c.OpenShiftHelper().InstallMetrics(f, openshift.MetricsHost(c.RoutingSuffix, c.ServerIP), c.Image, c.ImageVersion)
 }
 
-// CheckServiceCatalogPrereqVersion ensures the OpenShift server version is high enough to
-// run the service catalog.
-func (c *ClientStartConfig) CheckServiceCatalogPrereqVersion(out io.Writer) error {
```
Comment (mfojtik) [inline pkg/oc/bootstrap/docker/up.go]: shouldn't this be separate with SC removal?

**Comment by mfojtik** [comment]
one question, LGTM otherwise

File: pkg/oc/bootstrap/docker/up.go (hunk 2 above)
Comment (deads2k) [reply pkg/oc/bootstrap/docker/up.go]: > shouldn't this be separate with SC removal?

This is just a version check, not the SC installation.  We don't support running 3.7 with 3.9 or 3.10, so we don't need the check anymore.

**Pull request merged**